Library/Homebrew/cmd/doctor.rb

```diff
@@ -424,10 +424,18 @@ def check_access_share_man
     __check_subdir_access "share/man"
   end
 
+  def check_access_homebrew_repository
+    unless HOMEBREW_REPOSITORY.writable_real? then <<-EOS.undent
```

Comment:
I presume there's not some edge case where it's possible for someone to run `brew doctor` when `HOMEBREW_REPOSITORY` doesn't exist and see this?